cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/transform/MockPipelineConfigurer.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.mock.transform;
+
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.module.DatasetModule;
+import co.cask.cdap.api.plugin.PluginProperties;
+import co.cask.cdap.api.plugin.PluginSelector;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.StageConfigurer;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Mock test configurer used to test configure pipeline's validation of input schema and setting of output shcema.
+ */
+public class MockPipelineConfigurer implements PipelineConfigurer {
+  private static final String CORE_VALIDATOR = "core";
+  private final Schema inputSchema;
+  private Schema outputSchema;
+  private Map<String, Object> plugins;
+
+
+  public MockPipelineConfigurer(Schema inputSchema, Map<String, Object> plugins) {
+    this.inputSchema = inputSchema;
+    this.plugins = plugins;
+  }
+
+  @Nullable
+  public Schema getOutputSchema() {
+    return outputSchema;
+  }
+
+  @Override
+  public StageConfigurer getStageConfigurer() {
+    return new StageConfigurer() {
+      @Nullable
+      @Override
+      public Schema getInputSchema() {
+        return inputSchema;
+      }
+
+      @Override
+      public void setOutputSchema(@Nullable Schema schema) {
+        outputSchema = schema;
+      }
+    };
+  }
+
+  @Nullable
+  @Override
+  public <T> T usePlugin(String s, String s1, String s2, PluginProperties pluginProperties) {
+    return (T) plugins.get(s1);
```

Comment:
seems like its better to use the id as the key and not the name, since name is not unique. Can have the same name but a different type.